examples/scripts/gkd.py

```diff
@@ -0,0 +1,163 @@
+    --model_name_or_path="Qwen/Qwen2-0.5B-Instruct" \
+    --teacher_model_name_or_path="Qwen/Qwen2-1.5B-Instruct" \
+    --dataset_name="andito/chatbot_arena_completions" \
+    --report_to="wandb" \
+    --learning_rate=2e-5 \
+    --per_device_train_batch_size=16 \
+    --gradient_accumulation_steps=2 \
+    --output_dir="gkd-model" \
+    --logging_steps=1 \
+    --num_train_epochs=3 \
+    --push_to_hub \
+    --gradient_checkpointing
+
+# LoRA:
+python examples/scripts/gkd.py \
+    --model_name_or_path="Qwen/Qwen2-0.5B-Instruct" \
+    --teacher_model_name_or_path="Qwen/Qwen2-1.5B-Instruct" \
+    --dataset_name="andito/chatbot_arena_completions" \
+    --report_to="wandb" \
+    --learning_rate=2e-4 \
+    --per_device_train_batch_size=16 \
+    --gradient_accumulation_steps=2 \
+    --output_dir="gkd-model" \
+    --logging_steps=1 \
+    --num_train_epochs=3 \
+    --push_to_hub \
+    --gradient_checkpointing \
+    --use_peft \
+    --lora_r=64 \
+    --lora_alpha=16
+"""
+
+import logging
+import os
+from contextlib import nullcontext
+
+from datasets import load_dataset
+from tqdm.rich import tqdm
+from transformers import AutoTokenizer
+
+from trl import (
+    GKDConfig,
+    GKDTrainer,
+    ModelConfig,
+    RichProgressCallback,
+    get_kbit_device_map,
+    get_peft_config,
+    get_quantization_config,
+)
+from trl.commands.cli_utils import SFTScriptArguments, TrlParser, init_zero_verbose
+from trl.env_utils import strtobool
+from trl.trainer.callbacks import LogCompletionsCallback
+
+
+TRL_USE_RICH = strtobool(os.getenv("TRL_USE_RICH", "0"))
+
+if TRL_USE_RICH:
+    init_zero_verbose()
+    FORMAT = "%(message)s"
+
+    from rich.console import Console
+    from rich.logging import RichHandler
+
+    logging.basicConfig(format=FORMAT, datefmt="[%X]", handlers=[RichHandler()], level=logging.INFO)
+
+tqdm.pandas()
+
+
+if __name__ == "__main__":
+    parser = TrlParser((SFTScriptArguments, GKDConfig, ModelConfig))
+    args, training_args, model_config = parser.parse_args_and_config()
+
+    # Force use our print callback
+    if TRL_USE_RICH:
+        training_args.disable_tqdm = True
+        console = Console()
+
+    ################
+    # Model & Tokenizer
+    ################
+    quantization_config = get_quantization_config(model_config)
+    model_kwargs = dict(
+        revision=model_config.model_revision,
+        trust_remote_code=model_config.trust_remote_code,
+        attn_implementation=model_config.attn_implementation,
+        torch_dtype=model_config.torch_dtype,
+        use_cache=False if training_args.gradient_checkpointing else True,
+        device_map=get_kbit_device_map() if quantization_config is not None else None,
+        quantization_config=quantization_config,
+    )
+    training_args.model_init_kwargs = model_kwargs
+
+    tokenizer = AutoTokenizer.from_pretrained(
+        model_config.model_name_or_path,
+        trust_remote_code=model_config.trust_remote_code,
+        use_fast=True,
```

Comment:
Needed for correct generations:

```suggestion
        padding_side="left",
```